descwl_shear_sims/sim.py

```diff
@@ -319,7 +331,9 @@ def make_exp(
 
     shear = galsim.Shear(g1=g1, g2=g2)
     dims = [dim]*2
-    cen = (np.array(dims)-1)/2
+    # I think Galsim uses 1 offset. An array with length=dim=5
+    # The center is at 3=(5+1)/2
+    cen = (np.array(dims)+1)/2
```

Comment:
Galsim is 1-offset by defaulkt but the Rubin science pipelines use 0-offset coordinates. Given how the center is used below I am pretty sure 0-offset is correct here.

Reply:
Let's look into the details of the original code:
https://github.com/LSSTDESC/descwl-shear-sims/blob/5a693fd926f5e99ea1a0f76857dc66085e313ad5/descwl_shear_sims/sim.py#L322-L324
`cen` defined here adopts DM convention!
 
However, you can see the `cen` is first passed into Galsim and used to defined Galsim WCS:

https://github.com/LSSTDESC/descwl-shear-sims/blob/5a693fd926f5e99ea1a0f76857dc66085e313ad5/descwl_shear_sims/sim.py#L336-L342

After that, the Galsim WCS is used to define a DM WCS 
https://github.com/LSSTDESC/descwl-shear-sims/blob/5a693fd926f5e99ea1a0f76857dc66085e313ad5/descwl_shear_sims/sim.py#L395

and in the function here:
https://github.com/LSSTDESC/descwl-shear-sims/blob/5a693fd926f5e99ea1a0f76857dc66085e313ad5/descwl_shear_sims/wcs/dmwcs.py#L25-L29
The Galsim center is transformed to the DM center by doing ` Point2D(crpix[0]-1, crpix[1]-1)`. 
The problem is that at the beginning, the Galsim center does not use Galsim convention..

Reply:
Let's assume the `dim=33`, then we have `cen=16`
The Gaism image should have grids at 1,2,3...33. 
The correct Galsim Center should be at (17,17), however we will get an inconsistent one which is (16,16) 
Then, if we pass this inconsistent Galsim WCS to the function `make_dm_wcs()`, we will get 
`stack_crpix=Point2D(15,15)`, which is not correct.

Reply:
@rmjarvis Do what do you think of this, did we mess up the galsim WCS?

Reply:
I don't know.  I would write a unit test that checks that the connection between some particular pixel and its Ra/Dec is right.  I'm not following enough of what you are doing here to know whether you did everything right.  But presumably making a test like that should be straightforward.

Reply:
OK!! I will add a really simple test for one galaxy and force the offset to [0,0] to show that for the original code, the galaxy is not at the center of the image but has a (-1,-1) offset to the image center. I have done a similar test (but more complicated) in my shear estimation repo.. but I will put that one here..

Reply:
We have a test like this where we coadd psfs. Maybe we don't check the offset there though.